File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        for(int i=2; i<=n; i++){
            if(isPrime(i)){
                answer++;
            }
        }
        return answer;
    }
    
    public boolean isPrime(int num){
        if(num<2){
            return false;
        }
        for(int i=2; i<= (int)Math.sqrt(num); i++){
            if(num%i==0){
                return false;
            }
        }
        return true;
    }
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
import math

def isPrime(num):
    if num < 2:
        return False
    for i in range(2, int(math.sqrt(num))+1):
        if num%i==0:
            return False;
    return True;

def solution(numbers):
    answer = 0
    arr = list(map(int, numbers))
    tmp = []
    for i in range(1, len(arr)+1):
        tmp += permutations(numbers, i)
    #for t in tmp: print(t)
    #numlist = [''.join(t) for t in tmp]
    numlist = [int(''.join(t)) for t in tmp]
    ans = set()
    for n in numlist:
        if isPrime(n):
            #print(n)
            ans.add(n)
    return len(ans)
